src/services/address.service.ts

```diff
@@ -47,36 +101,143 @@ class AddressService {
 
+  public async exact(request: any): Promise<any> {
+    return new Promise<any>(async (resolve, reject) => {
+      if (!request || Object.keys(request.body).length === 0) {
+        loggerService.debug({ message: NULL_ADDRESS_REQUEST_ERROR }).flush();
+        reject(new Error(NULL_ADDRESS_REQUEST_ERROR));
+        return;
+      }
+
+      if (
+        Validator.isNotNullOrUndefined([
+          request.body.zipcode,
+          request.body.city,
+          request.body.state,
+          request.body.number,
+          request.body.street,
+        ])
+      ) {
+        loggerService.debug({ message: QUERY_MISSING_PARAMETERS }).flush();
+        reject(new Error(QUERY_MISSING_PARAMETERS));
+        return;
+      }
+
+      this.request(request)
+        .then((response) => {
+          const results: AddressResult[] = response;
+
+          const normalize = (str: string) => str.trim().toUpperCase();
+
+          const output = results.find((result) => {
+            return (
+              normalize(result.number) === normalize(request.body.number) &&
+              normalize(result.street) === normalize(request.body.street) &&
+              normalize(result.street2 || "") ===
+                normalize(request.body.street2 || "")
+            );
+          });
+
+          if (!output) {
+            reject(new Error(ADDRESS_NOT_FOUND_ERROR));
+            loggerService.warning({ message: ADDRESS_NOT_FOUND_ERROR }).flush();
+            return;
+          }
+
+          if (output.street2) {
+            output.formattedAddress = `${output.number} ${output.street} ${output.street2}, ${output.city}, ${output.state} ${output.zipcode}`;
+          } else {
+            output.formattedAddress = `${output.number} ${output.street}, ${output.city}, ${output.state} ${output.zipcode}`;
+          }
+
+          resolve(output);
+        })
+        .catch((error) => {
+          const formattedError =
+            "Exception Caught in address.service.ts -> \n exact -> \n this.request " +
+            error.message;
+          loggerService.warning({ message: formattedError }).flush();
+          reject(new Error(formattedError));
+          return;
+        });
+    });
+  }
+
   public async distance(addressRequest?: any): Promise<any> {
     // Complete this
+    return new Promise<any>(async (resolve, reject) => {
+      if (!addressRequest || Object.keys(addressRequest.body).length === 0) {
+        loggerService.debug({ message: NULL_ADDRESS_REQUEST_ERROR }).flush();
+        reject(new Error(NULL_ADDRESS_REQUEST_ERROR));
+        return;
+      }
+      try {
+        let newRequest = { body: addressRequest.body.addresses[0] };
+
+        let address1 = await this.exact(newRequest);
+        let address2 = await this.exact({
+          body: addressRequest.body.addresses[1],
+        });
+
+        this.getDistance(
+          address1.latitude,
+          address1.longitude,
+          address2.latitude,
+          address2.longitude
```

Comment:
You could clean this up by creating a custom type and passing that. e.g.,
`Type Location = {
  Latitude: number;
  Longitude: number;
};`

Then pass in two params of type Location.